utilities/rocAL/rocal_external_source/rocal_external_source.cpp

```diff
@@ -0,0 +1,382 @@
+
+    if(argc >= argIdx+MIN_ARG_COUNT)
+        processing_device = atoi(argv[++argIdx]);
+
+    if(argc >= argIdx+MIN_ARG_COUNT)
+        decode_width = atoi(argv[++argIdx]);
+
+    if(argc >= argIdx+MIN_ARG_COUNT)
+        decode_height = atoi(argv[++argIdx]);
+
+    if(argc >= argIdx+MIN_ARG_COUNT)
+        inputBatchSize = atoi(argv[++argIdx]);
+
+    if(argc >= argIdx+MIN_ARG_COUNT)
+        rgb = atoi(argv[++argIdx]);
+
+    if(argc >= argIdx+MIN_ARG_COUNT)
+        display = atoi(argv[++argIdx]);
+
+    if(argc >= argIdx+MIN_ARG_COUNT)
+        mode = atoi(argv[++argIdx]);
+
+    std::cerr<<"\n Mode:: "<< mode << std::endl;
+    std::cerr << ">>> Running on " << (processing_device?"GPU":"CPU") << std::endl;
+    RocalImageColor color_format = RocalImageColor::ROCAL_COLOR_RGB_PLANAR;
+    if (rgb == 0)
+      color_format = RocalImageColor::ROCAL_COLOR_U8;
+    else if (rgb == 1)
+      color_format = RocalImageColor::ROCAL_COLOR_RGB24;
+    else if (rgb == 2)
+      color_format = RocalImageColor::ROCAL_COLOR_RGB_PLANAR;
+
+    int channels = 3;
+    if (rgb == 0)
+        channels = 1;
+
+    auto handle = rocalCreate(inputBatchSize, processing_device?RocalProcessMode::ROCAL_PROCESS_GPU:RocalProcessMode::ROCAL_PROCESS_CPU, 0,1);
+
+    if(rocalGetStatus(handle) != ROCAL_OK) {
+        std::cerr << "Could not create the Rocal contex\n";
+        return -1;
+    }
+
+    /*>>>>>>>>>>>>>>>>>>> Graph description <<<<<<<<<<<<<<<<<<<*/
+    RocalImage input1;
+    std::vector<uint32_t> srcsize_height, srcsize_width;
+    uint32_t maxheight = 0, maxwidth = 0;
+    DIR *_src_dir;
+    struct dirent *_entity;
+    std::vector<std::string> file_names;
+    std::vector<unsigned char *> input_buffer;
+    if((_src_dir = opendir (folderPath1)) == nullptr) {
+            std::cerr<<"\n ERROR: Failed opening the directory at "<<folderPath1;
+            exit(0);
+    }
+
+    while((_entity = readdir (_src_dir)) != nullptr) {
+        if(_entity->d_type != DT_REG)
+            continue;
+
+        std::string file_path = folderPath1;
+        file_path.append(_entity->d_name);
+        file_names.push_back(file_path);
+    }
+    if(mode != 0) {
+        if(mode == 1){ // Raw compressed
+            srcsize_height.resize(file_names.size());
+            srcsize_width.resize(file_names.size());
+            for(uint32_t i = 0; i < file_names.size(); i++) {
+                FILE* _current_fPtr;
+                _current_fPtr = fopen(file_names[i].c_str(), "rb");// Open the file,
+                if(!_current_fPtr) // Check if it is ready for reading
+                    return 0;
+                fseek(_current_fPtr, 0 , SEEK_END);// Take the file read pointer to the end
+                size_t _current_file_size = ftell(_current_fPtr);// Check how many bytes are there between and the current read pointer position (end of the file)
+                unsigned char* input_data = (unsigned char *) malloc(sizeof(unsigned char ) * _current_file_size);
```

Comment:
Please use static_cast instead of c-style casting in this file
Applicable to all such instances in this file